sherpa/plot/__init__.py

```diff
@@ -63,6 +40,16 @@
 from sherpa.optmethods import LevMar, NelderMead
 from sherpa.stats import Likelihood, LeastSq, Chi2XspecVar
 from sherpa import get_config
+from sherpa.plot.backends import BaseBackend, BasicBackend, PLOT_BACKENDS
+# PLOT_BACKENDS only contains backends in modules that are imported successfully
+# but modules are not discovered by itself. Entrypoints would solve this problem
+# but the current implementation does not have this capability.
+# See docstring of sherpa.plot.backends.MetaBaseBackend for details.
+try:
+    import sherpa.plot.pylab_backend
+except ImportError:
+    pass
+
```

Comment:
There is something magical happening here: Backend classes are registered into PLOT_BACKENDS when the class object (not instance!) is generated, which calls the relevant functions in the metaclass. That happens, when the module that the defines the class is first imported. We want `'pylab'` to be in the list when a user does 
```python
from sherpa import plot
plot.set_backend('pylab')
```
Without this import here, the user would have to add the following line `import sherpa.plot.pylab_backend`, just to get the backend into the list, but the whole purpose if the `set_backend()` is to hide where and how exactly the backends are implemented.

The same applies to `BaseBackend` etc.

Reply:
So is the idea that we need to add a `try/import` for each new backend to `sherpa/plot/__init__.py`?

Or, for users who may want to write their own backend (and so can't change this file), they need to explicitly load the backend before `set_backend` will work.

Reply:
Yes, for now that's what we need to do. Entrypoints would solve that, but (as I said elsewhere) seem to be a pain to support for <3.10 and >=3.10 simultaneously. It's an extra three lines of code per backend that we ship and not worth any further magic (like auto importing based on a filename that starts with "backend_"). IF you have your own backend, you also still have the option to do:
```python
import my_backend
from sherpa import plot
plot.backend = my_backend.AwesomeBackend()
```
either way, I don't think it's unnatural that one has to import something before it can be used. Plus, I don't expect a proliferation of dozens of backends - I'd be surprised if we implement more than matplotlib, bokeh, ds9 (plus those used for testing and the example in the docs).